atc/exec/get_step.go

```diff
@@ -281,3 +313,51 @@ func (step *GetStep) run(ctx context.Context, state RunState, delegate GetDelega
 
 	return succeeded, nil
 }
+
+func (step *GetStep) getFromLocalCache(
+	logger lager.Logger,
+	teamId int,
+	resourceCache db.UsedResourceCache,
+	workerSpec worker.WorkerSpec) (worker.GetResult, bool, error) {
+	volume, found := step.findResourceCache(logger, teamId, resourceCache, workerSpec)
+	if !found {
+		return worker.GetResult{}, false, nil
+	}
+	metadata, err := step.resourceCacheFactory.ResourceCacheMetadata(resourceCache)
+	if err != nil {
+		return worker.GetResult{}, false, err
+	}
+	return worker.GetResult{
+		ExitStatus: 0,
+		VersionResult: runtime.VersionResult{
+			Version:  resourceCache.Version(),
+			Metadata: metadata.ToATCMetadata(),
+		},
+		GetArtifact: runtime.GetArtifact{volume.Handle()},
+	}, true, nil
+}
+
+func (step *GetStep) findResourceCache(
+	logger lager.Logger,
+	teamId int,
+	resourceCache db.UsedResourceCache,
+	workerSpec worker.WorkerSpec) (worker.Volume, bool) {
+	workers, err := step.workerPool.FindWorkersForResourceCache(logger, teamId, resourceCache.ID(), workerSpec)
+	if err != nil {
+		return nil, false
+	}
+
+	for _, sourceWorker := range workers {
+		volume, found, err := sourceWorker.FindVolumeForResourceCache(logger, resourceCache)
+		if err != nil {
+			logger.Debug("ignore-error", lager.Data{"error": err})
```

Comment:
Can we give a little more context for this log message? Either adding a `Session` (indicating that we're `finding-resource-cache`) or changing the message to something like `ignore-find-resource-cache-error`. Maybe we could also add the worker name in the `lager.Data`

Reply:
As we now mark streamed volume as resource cache, it may find multiple workers containing target resource cache. Then this `for` verify resource cache on each worker till find. A worker may crash (or other failures) in the same time, that should not break the current build, because other worker may also provide the resource cache. That's reason why we ignore the error here. I can revise the error message.